core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -611,7 +619,7 @@ class KafkaServer(
                 // ignore and try again
             }
           }
-          if (!shutdownSucceeded) {
+          if (!shutdownSucceeded && remainingRetries > 0) {
             Thread.sleep(config.controlledShutdownRetryBackoffMs)
             warn("Retrying controlled shutdown after the previous attempt failed...")
```

Comment:
As we expected that even there's no error returned, the shutdown might still not complete because there are some partitions remaining to move, not "failure" mentioned here in the warning message. Maybe we just need to log with `info` with `s"Retrying controlled shutdown with $remainingRetries retries left`.

Reply:
I do agree that the information is a bit redundant here. I was also wondering if we could combine this one into the previous ones. For instance, we could say: `Controlled shutdown request returned after XXXms with error ERROR. Retrying in XXXms ($remainingRetries retries left).`. The new part would be printed out only if remainingRetries > 0.

Reply:
I decided to leave this as a separate message since there are 3 cases where we might need to retry. However, I changed the level to INFO to be consistent and I added `remainingRetries` to the message as suggested.